src/web/components/MostViewed/MostViewedRight/MostViewedRightWrapper.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { css } from 'emotion';
+import { MostViewedRight } from './MostViewedRight';
+
+const flexGrow = css`
+    flex-grow: 1;
+`;
+
+interface Props {
+    limitItems?: number;
+}
+
+// Wrapping MostViewedRight so we can determine whether or not there's enough vertical space in the container to render it.
+export const MostViewedRightWrapper = ({ ...props }: Props) => {
+    const bodyRef = useRef<any>(null);
+    const [myHeight, setMyHeight] = useState(0);
+
+    useEffect(() => {
+        const { clientHeight } = bodyRef.current;
+        setMyHeight(clientHeight);
+    }, [myHeight]);
+
+    // Minimum height needed to render MostViewedRight is its own outer height.
+    const minWrapperHeight = 550;
+    return (
+        <div ref={bodyRef} className={flexGrow}>
+            {myHeight > minWrapperHeight ? (
+                <MostViewedRight {...props} />
+            ) : null}
```

Comment:
Could this be rewritten to use `&&`? I.e. `{myHeight > minWrapperHeight && (<MostViewedRight {...props} />)}` Only because I _think_ that's consistent with what I've seen elsewhere on the project (if not feel free to ignore!)

Reply:
Good point - I would agree with you if this wasn't the top level return statement from the component's render function. I reckon it's better to be explicit about it and return either `<MostViewedRight >` or null.

Reply:
I'm fairly sure the `{isSomeThing && <MyComp />}` pattern will return null. It's used throughout the codebase. We did discuss using `? A : null` but decided to go with the former

Reply:
I think the pattern (described [here](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)) will return the left hand side of the expression if it's falsey, otherwise the right hand side. According to that link, React will skip over anything which resolves to `false`. I do think it's different at the top-level, like @andre1050 said, in that we want to return `null`, but I'm not sure.

Reply:
So yes, both the inline if with else pattern and [the inline if with &&](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) work the same.

If the component is deciding what to return at the top level then you don't need to do anything inline and can just use pure javascript to decide what to return:

```
if (!someThing) return null;
else return (
   <div>{someThing}</div>
);

In general, the first pattern works well when there's no else component as it's cleaner and can prevent issues with typescript. It's very much a styling thing though 🤷‍♂